cylc/flow/workflow_files.py

```diff
@@ -1075,24 +1074,27 @@ def parse_reg(reg: str, src: bool = False) -> Tuple[str, Path]:
 
     Returns:
         reg: The normalised workflow arg.
-        path: (Only if src is True) The absolute path to the workflow file
-            (flow.cylc or suite.rc).
+        path: If src is True, the absolute path to the workflow file
+            (flow.cylc or suite.rc). Otherwise, the absolute path to the
+            workflow run dir.
     """
     if not src:
         validate_workflow_name(reg)
     reg: Path = Path(expand_path(reg))
 
     if src:
         reg, abs_path = _parse_src_reg(reg)
+        abs_file_path = abs_path
     else:
         abs_path = Path(get_workflow_run_dir(reg))
         if abs_path.is_file():
             raise WorkflowFilesError(
                 f"Workflow name must refer to a directory, not a file: {reg}"
             )
         abs_path, reg = infer_latest_run(abs_path)
+        abs_file_path = check_flow_file(abs_path, logger=None)
```

Comment:
Hmm, I think `check_flow_file()` will raise if no file is found. But `parse_reg()` shouldn't care whether the file is there or not if `src=False`, as used by "online" commands.

Reply:
Ah, good point. I'll rethink that.

Reply:
(I've rediscovered why I was getting double deprecation messages if `check_flow_file` does it - installation results in two checks, one for the installed file, one for the source file).

Reply:
Found a solution, see what you think.